### PR TITLE
fix!: normalize posqltimezone

### PR DESCRIPTION
**Pull request by JayWhite2357**
# Rationale for this change

The `PoSQLTimeZone` type has multiple representations for the same value, leading to complex upstream usage.

# What changes are included in this PR?

See the commit messages.

# Are these changes tested?
Yes, by existing, refactored, tests.

### CODE REVIEW

File: crates/proof-of-sql/src/sql/parse/query_context_builder.rs [hunk 1]
```diff
@@ -274,8 +274,8 @@ impl<'a> QueryContextBuilder<'a> {
 
 /// TODO: add docs
```
Comment (Dustin-Ray) [inline crates/proof-of-sql/src/sql/parse/query_context_builder.rs:275]: NIT: While we're here in this file, a doc comment might be nice

File: crates/proof-of-sql/src/sql/parse/query_context_builder.rs [hunk 2]
```diff
@@ -274,8 +274,8 @@ impl<'a> QueryContextBuilder<'a> {
...
 pub(crate) fn type_check_binary_operation(
-    left_dtype: &ColumnType,
-    right_dtype: &ColumnType,
```
Comment (Dustin-Ray) [inline crates/proof-of-sql/src/sql/parse/query_context_builder.rs:278]: NIT: Its not clear that these changes are related to the timestamp changes. This might be a better change in a separate PR.

Comment (JayWhite2357) [reply crates/proof-of-sql/src/sql/parse/query_context_builder.rs:278]: The changes to timestamp made `ColumnType` a smaller type, which made clippy request these changes.

Comment (Dustin-Ray) [reply crates/proof-of-sql/src/sql/parse/query_context_builder.rs:278]: Ok, feel free to resolve this NIT then.

**Comment by Dustin-Ray** [comment]
LGTM - Thank you

**Review by Dustin-Ray — APPROVED** [review]
LGTM
